sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/repository/query/ReactiveCosmosQueryExecution.java

```diff
@@ -54,6 +56,34 @@ public Object execute(DocumentQuery query, Class<?> type, String container) {
         }
     }
 
+    /**
+     * Find operation implementation to execute a find query for a single item
+     */
+    final class SingleEntityExecution implements ReactiveCosmosQueryExecution {
+
+        private final ReactiveCosmosOperations operations;
+        private final ReturnedType returnedType;
+
+        public SingleEntityExecution(ReactiveCosmosOperations operations, ReturnedType returnedType) {
+            this.operations = operations;
+            this.returnedType = returnedType;
+        }
+
+        @Override
+        public Object execute(DocumentQuery query, Class<?> type, String container) {
+            return operations.find(query, type, container)
+                .buffer(2)
```

Comment:
why do we need this `buffer(2)` ?

Reply:
We want to ensure only a single result is returned by the query. However, the results are returned as a stream. This allows us to check that the stream only contains a single item without retrieving all of the streams contents.

Reply:
Sounds good!